.github/workflows/r_nightly.yml

```diff
@@ -65,54 +68,58 @@ jobs:
           fi
           echo $PREFIX
 
-          archery crossbow download-artifacts -f r-nightly-packages -t binaries --skip-pattern-validation $PREFIX
+          archery crossbow download-artifacts -f r-nightly-packages -t binaries $PREFIX
+
+          if [ -n "$(ls -A binaries/*/*/)" ]; then
+            echo "Found files!"
+            echo "GOT_FILES=true" >> $GITHUB_ENV
+          else
+            echo "No files found. Stopping upload."
+            echo "GOT_FILES=false" >> $GITHUB_ENV
+          fi
       - name: Build Repository
+        if: env.GOT_FILES == 'true'
         shell: Rscript {0}

```

Comment:
Would it help with readability or maintainability to pull this script out to a standalone .R file, rather than have it inlined in the yaml?

Reply:
Up to you. I think this one is long enough to factor out, but you're going to be the one maintaining it, so if you prefer it this way, it's fine